### PR TITLE
New feature: styled strings

### PR DESCRIPTION
**Pull request by JelleInfinity**
* Added back escaped chars, placeholders and tags from previous version.
* Improved meta information
* Added special cases for styled tags

### CODE REVIEW

**Review by LuqueDaniel — CHANGES_REQUESTED** [review]
Nice feature 👍 

* In some cases the [interpolating data](https://www.renpy.org/doc/html/text.html#interpolating-data) isn't displayed well.
  imagen
* Is it intentional that tag attributes have the same style as strings?
  imagen

**Comment by JelleInfinity** [comment]
Oh yes, that's a difficult one.. I think the following pattern should be reasonable:

image



### This also catches missing brackets:



image
image



### With the exception of recursive missing brackets, where it catches the outer bracket now:



image

(Sorry about the large images)

**Comment by JelleInfinity** [comment]
As for the attribute style, it currently has a meta tag that doesn't override the style. I wasn't sure what to use there yet, and I think I want to add cases for known tags (like size and color) to make sure it displays correctly. Maybe for now I set it to embedded python code?

**Comment by JelleInfinity** [comment]
Also for know self-closing tags like image. The underline was kind of a debug style to notify me it was using the 'Start tag' fallback case

**Comment by JelleInfinity** [comment]
I'm working on some additions. We can have colored text like this:
image

But to make this available I need to make some modifications to the typescript parser, so it will take some time to add this.

I can also color the say statements with the character's color like that 😄

**Comment by LuqueDaniel** [comment]
@JelleInfinity Wow interesting! Let me know when it's ready for review. Or should we merge first?

**Comment by JelleInfinity** [comment]
Yeah I was already planning on adding that to the config. Also might need to think of some way to change the background color or something, in case someone has a color scheme that clashes with the vs code background color

**Comment by JelleInfinity** [comment]
I'd suggest you do a quick review again. I think this is almost ready :D

I'll add the custom scope generation in a different pull request

File: syntaxes/renpy.tmLanguage_2.json [hunk 1]
```diff
@@ -190,44 +171,378 @@
...
+
+    "string_tags": {
+      "patterns": [
+        {
+          "comment": "Valid tags without params (self-closing)",
+          "match": "({)\\s*(nw|done|fast|p|w|clear)\\s*(})",
+          "captures": {
+            "0": { "name": "meta.tag.$2.self-closing.renpy" },
+            "1": { "name": "punctuation.definition.tag.begin.renpy" },
+            "2": { "name": "entity.name.tag.$2.renpy" },
```
Comment (LuqueDaniel) [inline syntaxes/renpy.tmLanguage_2.json:269]: Interesting, I didn't know that this type of reference could be made. I like it!

**Review by LuqueDaniel — APPROVED** [review]
LGTM! 🎉 

@JelleInfinity  is ready to merge?

File: syntaxes/renpy.tmLanguage_2.json (hunk 1 above)
Comment (JelleInfinity) [reply syntaxes/renpy.tmLanguage_2.json:269]: Yeah, I got that from looking at the html plugin in the vs code [repo](https://github.com/microsoft/vscode/blob/1cca784e1fcd5144e29b0ee7da7d6a0e5dd00b1b/extensions/html/syntaxes/html.tmLanguage.json)

**Comment by JelleInfinity** [comment]
Yes, this is ready

**Pull request merged**